pkg/tfvars/gcp/gcp.go

```diff
@@ -78,5 +84,17 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 		cfg.PreexistingImage = false
 	}
 
+	if masterConfig.Disks[0].EncryptionKey != nil {
+		cfg.VolumeKMSKeyLink = generateDiskEncryptionKeyLink(masterConfig.Disks[0].EncryptionKey, masterConfig.ProjectID)
+	}
+
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+func generateDiskEncryptionKeyLink(keyRef *gcpprovider.GCPEncryptionKeyReference, projectID string) string {
```

Comment:
Maybe this belongs in GCP provider in a util package?

Reply:
To echo our conversation on slack, I would handle this in terraform:

1. create a type for the key similar to Auth in TFVars and pass the values in that way
2. conditionally create the self link variable (or null value therein) with logic similar to BYO networking: https://github.com/openshift/installer/blob/master/data/data/gcp/network/common.tf

I'm also fine with this approach and think it's OK to have this logic live here.